lib/functions/RegularFunctions.ts

```diff
@@ -185,6 +189,18 @@ const equality = {
       from: [ TypeURL.XSD_DATE_TIME, TypeURL.XSD_DATE_TIME ],
       to: [ TypeURL.XSD_DATE, TypeURL.XSD_DATE ],
     })
+    .set(
+      [ 'quad', 'quad' ],
+      context => ([ left, right ]) => {
+        const op: RegularFunction = new RegularFunction(RegularOperator.EQUAL, equality);
+        return bool(
+          (<E.BooleanLiteral> op.apply([ (<Quad> left).subject, (<Quad> right).subject ], context)).coerceEBV() &&
+          (<E.BooleanLiteral> op.apply([ (<Quad> left).predicate, (<Quad> right).predicate ], context)).coerceEBV() &&
+          (<E.BooleanLiteral> op.apply([ (<Quad> left).object, (<Quad> right).object ], context)).coerceEBV(),
+        );
+      },
+      false,
```

Comment:
The flag (when true = default value) makes sure that the current implementation throws an error when passed an invalid literal. (some functions don't want this, since they handle invalid literals themselves (see equality of terms))
I don't think this implementation will run into trouble so we can keep it as is.